File: frontend/src/types/recommendation.ts
export interface RankedApproach {
  approach: string;
  score: number;
  reason: string;
}

export interface SecondaryOption {
  approach: string;
  score: number;
  why: string[];
}

export interface MetadataEnrichmentSummary {
  status: "COLLECTED" | "PARTIAL" | "FAILED";
  source: import("./migration").OracleSourceMetadata | null;
  collected_fields: string[];
  applied_fields: string[];
  errors: string[];
  notes: string[];
}

export interface RecommendationResponse {
  request_id: string;
  recommended_approach: string;
  confidence: "HIGH" | "MEDIUM" | "LOW";
  score: number;
  why: string[];
  companion_tools: string[];
  prerequisites: string[];
  risk_flags: string[];
  secondary_option: SecondaryOption | null;
  rejected_approaches: RankedApproach[];
  manual_review_flags: string[];
  rules_version: string;
  metadata_enrichment?: MetadataEnrichmentSummary | null;
  generated_at: string;
}

File: backend/app/schemas/recommendation.py
from datetime import datetime, timezone
from typing import Literal

from pydantic import BaseModel, Field

from backend.app.schemas.oracle import MetadataEnrichmentSummary


class RankedApproach(BaseModel):
    approach: str
    score: int = Field(ge=0, le=100)
    reason: str


class SecondaryOption(BaseModel):
    approach: str
    score: int = Field(ge=0, le=100)
    why: list[str]


class RecommendationResponse(BaseModel):
    request_id: str
    recommended_approach: str
    confidence: Literal["HIGH", "MEDIUM", "LOW"]
    score: int = Field(ge=0, le=100)
    why: list[str]
    companion_tools: list[str]
    prerequisites: list[str]
    risk_flags: list[str]
    secondary_option: SecondaryOption | None = None
    rejected_approaches: list[RankedApproach]
    manual_review_flags: list[str]
    rules_version: str
    metadata_enrichment: MetadataEnrichmentSummary | None = None
    generated_at: datetime = Field(
        default_factory=lambda: datetime.now(timezone.utc),
    )
